File: VB/SchedulingDemo.Wpf/Data/SchedulerReportFactory.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.IO
Imports System.Linq
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Xpf.Scheduling
Imports DevExpress.Xpf.Scheduling.Reporting
Imports DevExpress.XtraScheduler.Reporting

Namespace SchedulingDemo
    Public Enum CalendarDetailPrintStyle
        <Display(Name := "Daily Style"), Image("pack://application:,,,/SchedulingDemo;component/Images/OutlookInspired/ReportTemplates/small_dayonepage.png")>
        DailyStyle
        <Display(Name := "Weekly Agenda Style"), Image("pack://application:,,,/SchedulingDemo;component/Images/OutlookInspired/ReportTemplates/small_weekonepage.png")>
        WeeklyAgendaStyle
        <Display(Name := "Weekly Calendar Style"), Image("pack://application:,,,/SchedulingDemo;component/Images/OutlookInspired/ReportTemplates/small_weeklefttorightonepage.png")>
        WeeklyCalendarStyle
        <Display(Name := "Monthly Calendar Style"), Image("pack://application:,,,/SchedulingDemo;component/Images/OutlookInspired/ReportTemplates/small_monthonepage.png")>
        MonthlyCalendarStyle
        <Display(Name := "Tri-fold Style"), Image("pack://application:,,,/SchedulingDemo;component/Images/OutlookInspired/ReportTemplates/small_trifold.png")>
        TriFoldStyle
    End Enum

    Public NotInheritable Class SchedulerReportFactory

        Private Sub New()
        End Sub
        Private Shared reportPathDictionary As Dictionary(Of CalendarDetailPrintStyle, String)

        Shared Sub New()
            Dim reportTemplates As List(Of ReportTemplateInfo) = Utils.GetReportTemplateCollection()
            reportPathDictionary = New Dictionary(Of CalendarDetailPrintStyle, String)()
            RegisterReport("DailyStyleFitToPage", CalendarDetailPrintStyle.DailyStyle, reportTemplates)
            RegisterReport("WeeklyStyle", CalendarDetailPrintStyle.WeeklyAgendaStyle, reportTemplates)
            RegisterReport("TrifoldStandard", CalendarDetailPrintStyle.TriFoldStyle, reportTemplates)
            RegisterReport("MonthlyStyle", CalendarDetailPrintStyle.MonthlyCalendarStyle, reportTemplates)
            RegisterReport("DailyStyleFitToPage", CalendarDetailPrintStyle.WeeklyCalendarStyle, reportTemplates)
        End Sub

        Public Shared Function Create(ByVal type As CalendarDetailPrintStyle, ByVal scheduler As SchedulerControl) As XtraSchedulerReport
            Dim reportPath As String = String.Empty
            If Not reportPathDictionary.TryGetValue(type, reportPath) Then
                Return Nothing
            End If
            Dim dateTimeRange As New DateTimeRange(scheduler.VisibleIntervals.First().Start, scheduler.VisibleIntervals.Last().End)
            Return Create(reportPath, scheduler, dateTimeRange)
        End Function
        Public Shared Function Create(ByVal reportTemplatePath As String, ByVal scheduler As SchedulerControl, ByVal dateTimeRange As DateTimeRange) As XtraSchedulerReport
            Dim report As New XtraSchedulerReport()
            scheduler.SchedulerPrintAdapter.DateTimeRange = dateTimeRange
            scheduler.SchedulerPrintAdapter.AssignToReport(report)
            report.LoadLayout(reportTemplatePath)
            Return report
        End Function

        Private Shared Sub RegisterReport(ByVal name As String, ByVal reportType As CalendarDetailPrintStyle, ByVal reportTemplates As IEnumerable(Of ReportTemplateInfo))
            Dim reportTemplate As ReportTemplateInfo = reportTemplates.FirstOrDefault(Function(x) Path.GetFileNameWithoutExtension(x.ReportTemplatePath) = name)
            If reportTemplate Is Nothing Then
                Return
            End If
            reportPathDictionary.Add(reportType, reportTemplate.ReportTemplatePath)
        End Sub
    End Class
End Namespace

File: CS/SchedulingDemo.Wpf/Data/SchedulerReportFactory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.IO;
using System.Linq;
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Xpf.Scheduling;
using DevExpress.Xpf.Scheduling.Reporting;
using DevExpress.XtraScheduler.Reporting;

namespace SchedulingDemo {
    public enum CalendarDetailPrintStyle {
        [Display(Name = "Daily Style"), Image("pack://application:,,,/SchedulingDemo;component/Images/OutlookInspired/ReportTemplates/small_dayonepage.png")]
        DailyStyle,
        [Display(Name = "Weekly Agenda Style"), Image("pack://application:,,,/SchedulingDemo;component/Images/OutlookInspired/ReportTemplates/small_weekonepage.png")]
        WeeklyAgendaStyle,
        [Display(Name = "Weekly Calendar Style"), Image("pack://application:,,,/SchedulingDemo;component/Images/OutlookInspired/ReportTemplates/small_weeklefttorightonepage.png")]
        WeeklyCalendarStyle,
        [Display(Name = "Monthly Calendar Style"), Image("pack://application:,,,/SchedulingDemo;component/Images/OutlookInspired/ReportTemplates/small_monthonepage.png")]
        MonthlyCalendarStyle,
        [Display(Name = "Tri-fold Style"), Image("pack://application:,,,/SchedulingDemo;component/Images/OutlookInspired/ReportTemplates/small_trifold.png")]
        TriFoldStyle
    }

    public static class SchedulerReportFactory {
        static Dictionary<CalendarDetailPrintStyle, string> reportPathDictionary;

        static SchedulerReportFactory() {
            List<ReportTemplateInfo> reportTemplates = Utils.GetReportTemplateCollection();
            reportPathDictionary = new Dictionary<CalendarDetailPrintStyle, string>();
            RegisterReport("DailyStyleFitToPage", CalendarDetailPrintStyle.DailyStyle, reportTemplates);
            RegisterReport("WeeklyStyle", CalendarDetailPrintStyle.WeeklyAgendaStyle, reportTemplates);
            RegisterReport("TrifoldStandard", CalendarDetailPrintStyle.TriFoldStyle, reportTemplates);
            RegisterReport("MonthlyStyle", CalendarDetailPrintStyle.MonthlyCalendarStyle, reportTemplates);
            RegisterReport("DailyStyleFitToPage", CalendarDetailPrintStyle.WeeklyCalendarStyle, reportTemplates);
        }

        public static XtraSchedulerReport Create(CalendarDetailPrintStyle type, SchedulerControl scheduler) {
            string reportPath = String.Empty;
            if (!reportPathDictionary.TryGetValue(type, out reportPath))
                return null;
            DateTimeRange dateTimeRange = new DateTimeRange(scheduler.VisibleIntervals.First().Start, scheduler.VisibleIntervals.Last().End);
            return Create(reportPath, scheduler, dateTimeRange);
        }
        public static XtraSchedulerReport Create(string reportTemplatePath, SchedulerControl scheduler, DateTimeRange dateTimeRange) {
            XtraSchedulerReport report = new XtraSchedulerReport();
            scheduler.SchedulerPrintAdapter.DateTimeRange = dateTimeRange;
            scheduler.SchedulerPrintAdapter.AssignToReport(report);
            report.LoadLayout(reportTemplatePath);
            return report;
        }

        static void RegisterReport(string name, CalendarDetailPrintStyle reportType, IEnumerable<ReportTemplateInfo> reportTemplates) {
            ReportTemplateInfo reportTemplate = reportTemplates.FirstOrDefault(x => Path.GetFileNameWithoutExtension(x.ReportTemplatePath) == name);
            if(reportTemplate == null)
                return;
            reportPathDictionary.Add(reportType, reportTemplate.ReportTemplatePath);
        }
    }
}
